File: app/src/main/java/com/castelijns/mmdemo/albums/AlbumsFragment.java
package com.castelijns.mmdemo.albums;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.castelijns.mmdemo.R;
import com.castelijns.mmdemo.app.BaseListFragment;
import com.castelijns.mmdemo.models.Album;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

import butterknife.BindView;

public class AlbumsFragment extends BaseListFragment implements AlbumsContract.View {

    @BindView(R.id.rv_albums)
    RecyclerView rvAlbums;

    private AlbumsContract.Presenter presenter;

    private AlbumsAdapter adapter;
    private List<Album> albums;
    private AlbumsAdapter.ItemClickListener itemClickListener;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        presenter = new AlbumsPresenter(this);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_albums, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        albums = new ArrayList<>();
        adapter = new AlbumsAdapter(getContext(), albums);
        adapter.setItemClickListener(itemClickListener);
        rvAlbums.setLayoutManager(new LinearLayoutManager(getContext()));
        rvAlbums.setHasFixedSize(true);
        rvAlbums.setAdapter(adapter);

        presenter.start();
    }

    @Override
    public void onDestroyView() {
        presenter.stop();
        super.onDestroyView();
    }

    public void setShowPhotosClickedListener(AlbumsAdapter.ItemClickListener listener) {
        itemClickListener = listener;
    }

    @Override
    public void showAlbums(List<Album> albums) {
        this.albums.addAll(albums);
        adapter.notifyDataSetChanged();
    }

    @Override
    public void showAlbumCount(int count) {
         setHeaderText(String.format(Locale.getDefault(), "%d %s", count, getString(R.string.header_albums)));
    }
}

File: app/src/main/kotlin/com/castelijns/mmdemo/albums/AlbumsFragment.kt
package com.castelijns.mmdemo.albums

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.castelijns.mmdemo.R
import com.castelijns.mmdemo.app.BaseListFragment
import com.castelijns.mmdemo.models.Album
import kotlinx.android.synthetic.main.fragment_albums.*
import java.util.*

class AlbumsFragment : BaseListFragment(), AlbumsContract.View {

    private lateinit var presenter: AlbumsPresenter

    private lateinit var adapter: AlbumsAdapter
    private var albums: MutableList<Album>? = null
    private var itemClickListener: ItemClickListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        presenter = AlbumsPresenter(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_albums, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        albums = mutableListOf()
        adapter = AlbumsAdapter(albums!!)
        adapter.setItemClickListener(itemClickListener!!)
        rv_albums.layoutManager = LinearLayoutManager(context)
        rv_albums.setHasFixedSize(true)
        rv_albums.adapter = adapter

        presenter.start()
    }

    override fun onDestroyView() {
        presenter.stop()
        super.onDestroyView()
    }

    fun setShowPhotosClickedListener(listener: ItemClickListener) {
        itemClickListener = listener
    }

    override fun showAlbums(albums: List<Album>) {
        this.albums!!.addAll(albums)
        adapter.notifyDataSetChanged()
    }

    override fun showAlbumCount(count: Int) {
        setHeaderText(getString(R.string.header_albums, count))
    }
}
